File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2)
{
    vector<string> answer;
    for (int i = 0; i < n; i++)
    {
        arr1[i] = arr1[i] | arr2[i];
        string ans = "";
        for (int j = 0; j < n; j++)
        {
            if (arr1[i] % 2 == 0)
                ans = " " + ans;
            else
                ans = "#" + ans;
            arr1[i] = arr1[i] >> 1;
        }
        answer.push_back(ans);
    }
    return answer;
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[arr1.length];
        
        for (int i = 0, idx = 0; i < arr1.length; i++) {
            int or = arr1[i] | arr2[i];
            
            String bitStr = Integer.toBinaryString(or)
                .replaceAll("1", "#")
                .replaceAll("0", " ");
            
            if (bitStr.length() < n) {
                String padding = " ".repeat(n - bitStr.length());
                bitStr = padding + bitStr;
            }
            
            answer[idx++] = bitStr;
        }
        return answer;
    }
}
//10110
//01110